### ISSUE TITLE
`alloc::realloc` does not follow requirements of `Layout::from_size_align_unchecked`

### ISSUE THREAD

**Issue by overlookmotel** [rust-lang/rust#108546]
The docs for `Layout::from_size_align_unchecked` state that:

> `size`, when rounded up to the nearest multiple of `align`, must not overflow `isize` (i.e., the rounded value must be less than or equal to `isize::MAX`)

`alloc::realloc`'s docs state a different constraint:

> `new_size`, when rounded up to the nearest multiple of `layout.align()`, must not overflow (i.e., the rounded value must be less than `usize::MAX`).

So this would be legal according to `alloc::realloc`'s docs:

```rust
let layout = alloc::Layout::from_size_align(1, 1);
let ptr1 = alloc::alloc(layout);
let ptr2 = alloc::realloc(ptr1, layout, usize::MAX);
```

But this results in a call internally:

```rust
Layout::from_size_align_unchecked(usize::MAX, 1)
```

...which is meant to be UB.

I assume this is just that the docs are out of date after #95295.

**Comment by thomcc** [comment]
Yes, this seems like a documentation bug to me.

**Comment by cuviper** [comment]
This was noted in https://github.com/rust-lang/rust/pull/95295#issuecomment-1078690691, discussed at length, and deferred in https://github.com/rust-lang/rust/pull/95295#issuecomment-1162604524: "This PR does not make a decision about `realloc`." There is also an ACP: https://github.com/rust-lang/libs-team/issues/61

**Comment by overlookmotel** [comment]
Apologies. I didn't read the whole (lengthy) thread. Just noticed the discrepancy in the docs, and was thoroughly confused why they didn't correspond.

Closing this in favour of https://github.com/rust-lang/libs-team/issues/61

I'll leave the PR open as I see you've tagged it "waiting for team", but feel free to just close it if you prefer. It's a trivial change anyway.
